src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -548,6 +548,7 @@ public void Commit(IReleaseSpec releaseSpec, IWorldStateTracer stateTracer, bool
                     FlushToTree();
                 }
 
+                codeFlushTask.GetAwaiter().GetResult();
```

Comment:
maybe instead of adding it here it should be at the end where it is now, but in finally block?

Reply:
I think its fine.  That causes many line of change.